src/Psd/PsdNativeFile_Linux.cpp

```diff
@@ -0,0 +1,164 @@
+
+bool NativeFile::DoOpenRead(const wchar_t* filename){
+    char *name = convert_ws(filename);
+    fd = open(name,O_RDONLY);
+    if(fd == -1){
+        PSD_ERROR("NativeFile","open(%s) => %s",name,strerror(errno));
+        std::free(name);
+        return false;
+    }
+    std::free(name);
+    return true;
+}
+bool NativeFile::DoOpenWrite(const wchar_t* filename){
+    char *name = convert_ws(filename);
+    //Create a new file
+    fd = open(name,O_WRONLY | O_CREAT,S_IRUSR | S_IWUSR);
+    if(fd == -1){
+        PSD_ERROR("NativeFile","open(%s) => %s",name,strerror(errno));
+        std::free(name);
+        return false;
+    }
+    std::free(name);
+    return true;
+}
+bool NativeFile::DoClose(){
+    int ret = close(fd);
+    fd = -1;
+    return ret == 0;
+}
+
+//Wrtie / Read
+
+File::ReadOperation NativeFile::DoRead(void* buffer, uint32_t count, uint64_t position){
+    aiocb *operation = memoryUtil::Allocate<aiocb>(m_allocator);
+    std::memset(operation,0,sizeof(aiocb));
+
+    operation->aio_buf = buffer;
+    operation->aio_fildes = fd;
+    operation->aio_lio_opcode = LIO_READ;//Do read
+    operation->aio_nbytes = count;
+    operation->aio_offset = position;
+    operation->aio_reqprio = 0;
+    operation->aio_sigevent.sigev_notify = SIGEV_NONE;//No signal will be send 
+
+    //OK Execute it
+    if(aio_read(operation) == -1){
+        //Has Error
+        PSD_ERROR("NativeFile","On DoRead aio_read() => %s",strerror(errno));
+        memoryUtil::Free(m_allocator,operation);
+        return nullptr;
+    }
+    return operation;
+}
+File::ReadOperation NativeFile::DoWrite(const void* buffer, uint32_t count, uint64_t position){
+    aiocb *operation = memoryUtil::Allocate<aiocb>(m_allocator);
+    std::memset(operation,0,sizeof(aiocb));
+    
+    operation->aio_buf = const_cast<void*>(buffer);
+    operation->aio_fildes = fd;
+    operation->aio_lio_opcode = LIO_WRITE;//Do Write
+    operation->aio_nbytes = count;
+    operation->aio_offset = position;
+    operation->aio_reqprio = 0;
+    operation->aio_sigevent.sigev_notify = SIGEV_NONE;//No signal will be send 
+
+    //OK Execute it
+    if(aio_write(operation) == -1){
+        //Has Error
+        PSD_ERROR("NativeFile","On DoWrite aio_read() => %s",strerror(errno));
+        memoryUtil::Free(m_allocator,operation);
+        return nullptr;
+    }
+    return operation;
+}
+
+//Wait for R/W
+
+static bool generic_wait(aiocb *operation,Allocator *alloc){
+    //Wait for it
+    if(aio_suspend(&operation,1,nullptr) == -1){
+        PSD_ERROR("NativeFile","aio_suspend() => %s",strerror(errno));
+        memoryUtil::Free(alloc,operation);
+        return false;
+    }
+    //Get status
+    ssize_t ret = aio_return(operation);
+    int errcode = aio_error(operation);
+
+    if(ret == -1){
+        PSD_ERROR("NativeFile","aio_error() %d => %s",errcode,strerror(errcode));
+    }
+
+    memoryUtil::Free(alloc,operation);
+    return ret != -1;
+}
```

Comment:
Helper functions needed only in one translation unit should go at the top of the .cpp, into an anonymous namespace.